File: DFS & BFS/BOJ_1260_DFS와 BFS.py
from collections import deque

def dfs(graph, v):
    dfs_visit[v] = True
    print(v, end=' ')
    for i in range(1, len(graph)):
        if (not dfs_visit[i]) and graph[v][i] == 1:
            dfs(graph, i)

def bfs(graph, start):
    queue = deque([start])
    bfs_visit[start] = True
    while queue:
        v = queue.popleft()
        print(v, end=' ')
        for i in range(1, len(graph)):
            if (not bfs_visit[i]) and graph[v][i] == 1:
                queue.append(i)
                bfs_visit[i] = True

N, M, V = map(int, input().split())
graph = [[0] * (N + 1) for _ in range(N + 1)]
dfs_visit = [False] * (N + 1)
bfs_visit = [False] * (N + 1)

for _ in range(M):
    row, col = map(int, input().split())
    graph[row][col] = graph[col][row] = 1

dfs(graph, V)
print()
bfs(graph, V)

File: DFS & BFS/BOJ_1260_DFS와 BFS.java
import java.io.*;
import java.util.*;

public class Main {

    static ArrayList<Integer>[] graph;
    static boolean[] visit;

    public static void main(String[] args) throws IOException {

        int n, m, v;

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st;

        st = new StringTokenizer(br.readLine(), " ");
        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        v = Integer.parseInt(st.nextToken());

        graph = new ArrayList[n + 1];
        for(int i = 0; i < n + 1; i++) {
            graph[i] = new ArrayList<Integer>();
        }

        int r, c;
        for(int mm = 0; mm < m; mm++) {
           st = new StringTokenizer(br.readLine(), " ");
           r = Integer.parseInt(st.nextToken());
           c = Integer.parseInt(st.nextToken());

           graph[r].add(c);
           graph[c].add(r);
        }

        for(int i = 1; i < n + 1; i++) {
            Collections.sort(graph[i]);
        }

        visit = new boolean[n + 1];
        dfs(v);
        System.out.println();

        Arrays.fill(visit, false);
        bfs(v);
        System.out.println();
    }

    private static void dfs(int v) {
        if(visit[v]) {
            return;
        }

        visit[v] = true;
        System.out.print(v + " ");

        for(int next : graph[v]) {
            if(!visit[next]) {
                dfs(next);
            }
        }
    }

    private static void bfs(int v) {
        Queue<Integer> q = new LinkedList<>();
        q.add(v);
        visit[v] = true;

        while(!q.isEmpty()) {
            int now = q.poll();
            System.out.print(now + " ");

            for(int next : graph[now]) {
                if(!visit[next]) {
                    q.add(next);
                    visit[next] = true;
                }
            }
        }
    }
}
